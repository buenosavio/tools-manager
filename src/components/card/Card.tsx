import { useState } from 'react'
import {
  CardItem,
  ItemDetails,
  RemoveButton,
  TagsList,
  Title,
  Description
} from './Card.styles'
import RemoveCard from '../remove-tool/RemoveTool'

const Card = ({ item }: any) => {
  const [modalOpen, setModalOpen] = useState<boolean>()

  return (
    <CardItem>
      <ItemDetails>
        <Title>{item.name}</Title>
        <Description>{item.description}</Description>
        <TagsList>
          {/* {item.tag[item.tag.length - 1].map((tag: any) => (
            <div key={tag.value}>
              <p>#{tag.label}</p>
            </div>
          ))} */}
        </TagsList>
      </ItemDetails>
      <RemoveButton
        id="remove-btn"
        onClick={() => {
          setModalOpen(true)
        }}
      >
        remove
      </RemoveButton>
      {modalOpen && <RemoveCard handleOpen={setModalOpen} item={item} />}
    </CardItem>
  )
}

export default Card
