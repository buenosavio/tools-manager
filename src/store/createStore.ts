/* eslint-disable import/no-anonymous-default-export */
import { createStore, applyMiddleware, Reducer, Store } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { routerMiddleware } from 'connected-react-router';
import { SagaMiddleware } from 'redux-saga';
import { History } from 'history';

const isDevelopMode = process.env.NODE_ENV === 'development';

export default (reducers: Reducer, sagaMiddleware: SagaMiddleware, history: History): Store => {
  if (isDevelopMode) {
    const storeEnhacer = composeWithDevTools({ trace: true, traceLimit: 30 })(
      applyMiddleware(sagaMiddleware, routerMiddleware(history))
    );
    return createStore(reducers, storeEnhacer);
  }
  return createStore(reducers, applyMiddleware(sagaMiddleware, routerMiddleware(history)));
};
