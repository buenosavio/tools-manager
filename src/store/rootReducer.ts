import ToolsReducer from './tools/duck'
import {combineReducers} from '@reduxjs/toolkit'

const rootReducer = combineReducers({
  ToolsReducer
})


export type AppState = ReturnType<typeof rootReducer>

export default rootReducer