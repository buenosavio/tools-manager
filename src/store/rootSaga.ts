import {all} from 'redux-saga/effects'
import toolSaga from './tools/saga'

export default function* rootSaga(): Generator {
  yield all([toolSaga()])
}