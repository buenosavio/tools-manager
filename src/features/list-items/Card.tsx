import { useState } from 'react'
import { CardItem, ItemDetails, RemoveButton } from './Card.styles'
import RemoveCard from './ModalRemoveCard'

const Card = ({ item }: any) => {
  const [modalOpen, setModalOpen] = useState<boolean>()
  return (
    <CardItem>
      <ItemDetails>
        <h4>{item.name}</h4>
        <p>{item.description}</p>
        {item.tag.map((tag: any) => (
          <div key={tag.id}>
            <p>#{tag.name}</p>
          </div>
        ))}
      </ItemDetails>
      <RemoveButton
        onClick={() => {
          setModalOpen(true)
        }}
      >
        remove
      </RemoveButton>
      {modalOpen && <RemoveCard handleOpen={setModalOpen} item={item} />}
    </CardItem>
  )
}

export default Card
