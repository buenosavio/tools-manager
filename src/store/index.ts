import createSagaMiddleware from "redux-saga";
import createStore from "./createStore";
import rootReducer from "./rootReducer";
import history from "./history";
import { persistReducer } from "redux-persist";
import persistStore from "redux-persist/es/persistStore";
import rootSaga from "./rootSaga";
import storage from 'redux-persist/lib/storage'

const sagaMiddleware = createSagaMiddleware()

const store = createStore(persistReducer({key: 'store', storage },rootReducer), sagaMiddleware, history);

const persistor = persistStore(store);

sagaMiddleware.run(rootSaga)

export type AppDispatch = typeof store.dispatch;

export {store, persistor}