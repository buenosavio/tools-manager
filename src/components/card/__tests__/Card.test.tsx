import withProvider from '../../../utils/withContext'
import renderer from 'react-test-renderer'
import Card from '../Card'
import { store } from '../../../store'
import { Provider } from 'react-redux'
import { mount } from 'enzyme'

describe('Card', () => {
  const tool = {
    id: '1',
    name: 'Tool 1',
    link: 'www.tools.com',
    description: 'description of my tool',
    tag: [
      { id: '1', name: 'tag 1', label: 'label1' },
      { id: '2', name: 'tag 1', label: 'label1' },
      { id: '3', name: 'tag 1', label: 'label1' }
    ]
  }
  const Component = withProvider(Card, store)
  test('Snapshot of component <Card />', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Card item={tool} />
      </Provider>
    )
    expect(tree).toMatchSnapshot()
  })
  test('Click to open modal', () => {
    const componentRedered = mount(<Component item={tool} />)
    componentRedered.find('#remove-btn').at(0).simulate('click')
    expect(componentRedered).toHaveLength(1)
  })
})
