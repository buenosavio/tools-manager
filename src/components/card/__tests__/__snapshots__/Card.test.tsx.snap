// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Snapshot of component <Card /> 1`] = `
.c0 {
  position: relative;
  background: #fdfdfd;
  box-shadow: 5px 6px 7px rgba(0,0,0,0.25);
  border: 1px solid #764e9e2c;
  border-radius: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 15px 30px;
  margin-bottom: 30px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 100%;
}

.c5 {
  position: absolute;
  top: 10%;
  left: 92%;
  width: 98px;
  background: none;
  border: none;
  background-image: url(removeIcon.svg);
  background-repeat: no-repeat;
  background-position: 0% 50%;
  font-size: 18px;
}

.c5:hover {
  cursor: pointer;
}

.c4 {
  min-width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  font-size: 18px;
  font-weight: 700;
  color: #170c3a;
}

.c2 {
  font-weight: 400;
  font-size: 24px;
  color: #170c3a;
}

.c3 {
  font-weight: 400;
  font-size: 18px;
  color: #170c3a;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <p
      className="c2"
    >
      Tool 1
    </p>
    <p
      className="c3"
    >
      description of my tool
    </p>
    <div
      className="c4"
    />
  </div>
  <button
    className="c5"
    id="remove-btn"
    onClick={[Function]}
  >
    remove
  </button>
</div>
`;
